OneSignalSDK/onesignal/src/main/java/com/onesignal/OSNotificationWorkManager.java

```diff
@@ -0,0 +1,91 @@
+package com.onesignal;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.work.Data;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkManager;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+class OSNotificationWorkManager {
+
+    private static final String ANDROID_NOTIF_ID_WORKER_DATA_PARAM = "android_notif_id";
+    private static final String JSON_PAYLOAD_WORKER_DATA_PARAM = "json_payload";
+    private static final String TIMESTAMP_WORKER_DATA_PARAM = "timestamp";
+    private static final String IS_RESTORING_WORKER_DATA_PARAM = "is_restoring";
+
+    static void beginEnqueueingWork(Context context, int androidNotificationId, String jsonPayload, boolean isRestoring, long timestamp, boolean isHighPriority) {
+        // TODO: Do we need isHighPriority only more?
+
+        Data inputData = new Data.Builder()
+                .putInt(ANDROID_NOTIF_ID_WORKER_DATA_PARAM, androidNotificationId)
+                .putString(JSON_PAYLOAD_WORKER_DATA_PARAM, jsonPayload)
+                .putLong(TIMESTAMP_WORKER_DATA_PARAM, timestamp)
+                .putBoolean(IS_RESTORING_WORKER_DATA_PARAM, isRestoring)
+                .build();
+
+        OneTimeWorkRequest workRequest = new OneTimeWorkRequest.Builder(NotificationWorker.class)
```

Comment:
Can it happend that for the same notification we call more than once this method?

Reply:
Not sure but its a good point so a good idea would be to use the OS id as a tag for the job and make sure a job does not exist with that id name already. Otherwise im pretty sure we check for notifications being duplicate before this line of code

Reply:
Here is my solution to what you are brining up, but maybe we want to change this or modify... Just an idea for now though...
```Java
WorkManager.getInstance(context)
    .enqueueUniqueWork(osNotificationId, ExistingWorkPolicy.KEEP, workRequest);
```

What does `ExistingWorkPolicy.KEEP` do? 

Well... If there is existing pending (uncompleted) work with the same unique name (in this case, OneSignal notification id), do nothing. Otherwise, insert the newly-specified work.